cmd/deployment/create.go

```diff
@@ -0,0 +1,169 @@
+{
+    "name": "my example cluster",
+    "resources": {
+        "apm": [
+            {
+                "display_name": "my apm instance",
+                "ref_id": "my-apm-instance",
+                "elasticsearch_cluster_ref_id": "my-es-cluster",
+                "plan": {
+                    "zone_count": 1,
+                    "apm": {
+                        "version": "6.8.4"
+                    },
+                    "size":{
+                        "resource": "memory",
+                        "value": 512
+                    }
+                }
+            }
+        ],
+        "elasticsearch": [
+            {
+                "display_name": "my elasticsearch cluster",
+                "ref_id": "my-es-cluster",
+                "plan": {
+                    "deployment_template": {
+                        "id": "default"
+                    },
+                    "elasticsearch": {
+                        "version": "6.8.4"
+                    },
+                    "cluster_topology": [
+                        {
+                            "instance_configuration_id": "data.default",
+                            "memory_per_node": 1024,
+                            "node_count_per_zone": 1,
+                            "node_type": {
+                                "data": true,
+                                "ingest": true,
+                                "master": true,
+                                "ml": false
+                            },
+                            "zone_count": 1
+                        }
+                    ]
+                }
+            }
+        ],
+        "kibana": [
+            {
+                "display_name": "my kibana instance",
+                "ref_id": "my-kibana-instance",
+                "elasticsearch_cluster_ref_id": "my-es-cluster",
+                "plan": {
+                    "zone_count": 1,
+                    "kibana": {
+                        "version": "6.8.4"
+                    },
+                    "cluster_topology": [
+                        {
+                            "instance_configuration_id": "kibana",
+                            "memory_per_node": 1024,
+                            "node_count_per_zone": 1
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+}
+$ ecctl deployment create -f deployment_example.json --version=7.4.1
+[...]`[1:]
+
+var createCmd = &cobra.Command{
+	Use:     `create -f <file definition.json>`,
+	Short:   "Creates a deployment from a file definition, allowing certain flag overrides",
+	Long:    createLong,
+	Example: createExample,
+	PreRunE: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		filename, _ := cmd.Flags().GetString("file")
+		var r models.DeploymentCreateRequest
+		if err := cmdutil.DecodeFile(filename, &r); err != nil {
+			return err
+		}
+
+		var region string
+		if ecctl.Get().Config.Region == "" {
```

Comment:
In the ECE world, there aren't regions, necessarily, at least there's no concept of federation or multiple regions in an ECE installation, as such, many if not all users won't necessarily be aware that the _default_ region is `ece-region` and thus, not have the `region` global configuration parameter set to any value, meaning empty.

If that is the case, I'm automatically making it `ece-region` as a default, and that will get also populated down the line to the deployment definition for each of the deployment's defined resources.

Reply:
Yeah I understand that. My question is why are not doing anything similar [here](https://github.com/elastic/ecctl/blob/master/cmd/deployment/delete.go#L33) for instance? I'm pointing to this particular command because it's on the same API endpoint and was implemented recently

Reply:
I mean if we want to use a default region this totally fine, but I think we need to be consistent across all commands or at least keep it on all new commands implemented from now on

Reply:
We chatted on slack and I was confused, assuming that `region` is needed as part of every API call which is not true at all. So my previous comments are note valid. Thanks @marclop for clearing this out !